builder/vsphere/common/step_import_to_content_library.go

```diff
@@ -0,0 +1,127 @@
+//go:generate struct-markdown
+//go:generate mapstructure-to-hcl2 -type ContentLibraryDestinationConfig
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/packer/builder/vsphere/driver"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+	"github.com/hashicorp/packer/template/interpolate"
+	"github.com/vmware/govmomi/vapi/vcenter"
+)
+
+// With this configuration Packer creates a library item in a content library whose content is a virtual machine template created from the just built VM.
+// The virtual machine template is stored in a newly created library item.
+type ContentLibraryDestinationConfig struct {
+	// Name of the library in which the new library item containing the VM template should be created.
+	// The Content Library should be of type Local to allow deploying virtual machines.
+	Library string `mapstructure:"library"`
+	// Name of the library item that will be created. The name of the item should be different from [vm_name](#vm_name).
+	// Defaults to [vm_name](#vm_name) + timestamp.
+	Name string `mapstructure:"name"`
+	// Description of the library item that will be created. Defaults to "Packer imported [vm_name](#vm_name) VM template".
+	Description string `mapstructure:"description"`
+	// Cluster onto which the virtual machine template should be placed.
+	// If cluster and resource_pool are both specified, resource_pool must belong to cluster.
+	// If cluster and host are both specified, host must be a member of cluster.
+	// Defaults to [cluster](#cluster).
+	Cluster string `mapstructure:"cluster"`
+	// Virtual machine folder into which the virtual machine template should be placed.
+	// Defaults to the same folder as the source virtual machine.
+	Folder string `mapstructure:"folder"`
+	// Host onto which the virtual machine template should be placed.
+	// If host and resource_pool are both specified, resource_pool must belong to host.
+	// If host and cluster are both specified, host must be a member of cluster.
+	// Defaults to [host](#host).
+	Host string `mapstructure:"host"`
+	// Resource pool into which the virtual machine template should be placed.
+	// Defaults to [resource_pool](#resource_pool). if [resource_pool](#resource_pool) is also unset,
+	// the system will attempt to choose a suitable resource pool for the virtual machine template.
+	ResourcePool string `mapstructure:"resource_pool"`
+	// The datastore for the virtual machine template's configuration and log files.
+	// Defaults to the storage backing associated with the library specified by library.
+	Datastore string `mapstructure:"datastore"`
+}
+
+func (c *ContentLibraryDestinationConfig) Prepare(lc *LocationConfig) []error {
+	var errs *packer.MultiError
+
+	if c.Library == "" {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("a library name must be provided"))
+	}
+	if c.Name != "" && c.Name == lc.VMName {
```

Comment:
No, lc.VMName is a required field. But I think your suggestion is right anyways because if the Name is empty then will never be equal to VMname 🤔  so is an unnecessary check

Reply:
Sorry for the confusion, I meant to say `if no` not  `if yes` :smile: :+1: